src/main/java/com/it4045/pokemonviewer/enterprise/dto/Pokemon.java

```diff
@@ -2,35 +2,48 @@
 
 import lombok.Data;
 
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
 
 public @Data
 class Pokemon {
 
-    private int pokemonId;
-    private String pokemonNumber;
-    private String pokemonName;
-    private String pokemonImg;
-    private List<String> pokemonType;
-    private String pokemonHeight;
-    private String pokemonWeight;
-    private String pokemonCandy;
-    private String pokemonEgg;
-    private double pokemonSpawnChoice;
-    private double pokemonAvgSpawns;
-    private String pokemonSpawnTime;
-    private List<Double> pokemonMultipliers;
-    private List<String> pokemonWeakness;
-    private List<Evolution> pokemonNextEvolution;
-    private List<Evolution> pokemonPrevEvolution;
+    private int id;
+    private String number;
+    private String name;
+    private String img;
+    private List<String> type;
+    private String height;
+    private String weight;
+    private String candy;
+    private String egg;
+    private double spawnChoice;
+    private double avgSpawns;
+    private String spawnTime;
+    private List<Double> multipliers;
+    private List<String> weakness;
+    private List<Map.Entry<Pokemon, String>> previousEvolution;
+    private List<Map.Entry<Pokemon, String>> nextEvolution;
```

Comment:
Second technical suggestion: combine the Evolution class into the Pokemon class.

While it can be a good idea to separate large classes into several smaller ones, I think that information about a Pokemon's evolution is mainly relevant within the Pokemon class. I had trouble deciding whether it would be better to combine the Evolution class into the Pokemon class, or to make an EvolutionFamily class that would contain all of the Pokemon in an evolution family and the ways in which they evolved, but I ended up choosing the former. This method would be rather tedious to fill in data for by hand, and it involves some duplication of data that an EvolutionFamily class would prevent, but since you are pulling data from a JSON API, I think that it may actually be the more simple and straightforward option.

Here, evolutions are stored as a list of paired Pokemon and String values. The String value refers to the method by which a Pokemon evolves. So, for example, an instance of the Pokemon class that represents the Pokemon "Eevee" might have several entries in the nextEvolution list: Flareon and "Fire Stone", Jolteon and "Thunder Stone", etc.